geoai/detectron2.py

```
@@ -240,18 +240,32 @@ def create_probability_mask(masks: np.ndarray, scores: np.ndarray) -> np.ndarray
     Returns:
         Probability mask with maximum confidence score for each pixel
     """
-    if len(masks) == 0:
-        return np.zeros((masks.shape[1], masks.shape[2]), dtype=np.float32)
+    if masks.ndim != 3:
+        raise ValueError(f"Expected masks shape (N, H, W), got {masks.shape}")
+
+    if scores.ndim != 1:
+        raise ValueError(f"Expected socres shape (N,) got {scores.shape}")
```

Ruff (0.15.15)

[warning] Avoid specifying long messages outside the exception class

(TRY003)